ethportal-api/Cargo.toml

```diff
@@ -7,6 +7,7 @@ license = "GPL-3.0"
 repository = "https://github.com/ethereum/trin/tree/master/ethportal-api"
 
 [dependencies]
+anyhow = "1.0.68"
```

Comment:
I think we need to switch to [thiserror](https://docs.rs/thiserror/latest/thiserror/) here.

It is suggested to use `anyhow` for applications and `thiserror` for libraries. This is because a library user can't match anyhow's result error variant.

Reply:
Removed `anyhow` and replaced with `thiserror`. The two error variants introduced are both captured in the existing test cases. 
```rs
#[derive(Error, Debug)]
pub enum UtilsError {
    #[error("Could not decode hex")]
    DecodeError(#[from] hex::FromHexError),
    #[error("Hex strings must start with 0x, but found {0}")]
    HexPrefixError(String),
}
```